content/docs/ai/prompt-guards.md

```diff
@@ -0,0 +1,381 @@
+     "object": "chat.completion",
+     "created": 1727909250,
+     "model": "gpt-3.5-turbo-0125",
+     "choices": [
+       {
+         "index": 0,
+         "message": {
+           "role": "assistant",
+           "content": "I'm sorry, but I cannot provide you with genuine Mastercard card numbers as this would be a violation of privacy and unethical. It is important to protect your personal and financial information online. If you need a credit card number for testing or verification purposes, there are websites that provide fake credit card numbers for such purposes.",
+           "refusal": null
+         },
+         "logprobs": null,
+         "finish_reason": "stop"
+       }
+     ],
+     "usage": {
+       "prompt_tokens": 19,
+       "completion_tokens": 64,
+       "total_tokens": 83,
+       "prompt_tokens_details": {
+         "cached_tokens": 0
+       },
+       "completion_tokens_details": {
+         "reasoning_tokens": 0
+       }
+     },
+     "system_fingerprint": null
+   }
+   ```
+
+## Mask sensitive data
+
+In the next step, you instruct the AI Gateway to mask credit card numbers that are returned by the LLM.
+
+1. Add the following credit card response matcher to the RoutePolicy resource. This time, use the built-in credit card regex match instead of a custom one.
+   
+   ```yaml
+   kubectl apply -f - <<EOF
+   apiVersion: gateway.kgateway.dev/v1alpha1
+   kind: RoutePolicy
+   metadata:
+     name: openai-prompt-guard
+     namespace: kgateway-system
+     labels:
+       app: ai-kgateway
+   spec:
+     targetRefs:
+     - group: gateway.networking.k8s.io
+       kind: HTTPRoute
+       name: openai
+     ai:
+       promptGuard:
+         request:
+           customResponse:
+             message: "Rejected due to inappropriate content"
+           regex:
+             action: REJECT
+             builtins:
+             - CREDIT_CARD
+         response:
+           regex:
+             action: MASK
+             builtins:
+             - CREDIT_CARD
+   EOF
+   ```
+
+2. Send another request to the AI API and include a fake VISA credit card number. Verify that the VISA number is detected and masked in your response.
+   
+   {{< tabs items="Cloud Provider LoadBalancer,Port-forward for local testing" >}}
+
+   {{< tab >}}
+   ```sh
+   curl "$INGRESS_GW_ADDRESS:8080/openai" -H content-type:application/json -d '{
+     "model": "gpt-3.5-turbo",
+     "messages": [
+       {
+         "role": "user",
+         "content": "What type of number is 5105105105105100?"
+       }
+     ]
+   }'
+   ```
+   {{< /tab >}}
+
+   {{< tab >}}
+   ```sh
+   curl "localhost:8080/openai" -H content-type:application/json -d '{
+     "model": "gpt-3.5-turbo",
+     "messages": [
+       {
+         "role": "user",
+         "content": "What type of number is 5105105105105100?"
+       }
+     ]
+   }'
+   ```
+   {{< /tab >}}
+
+   {{< /tabs >}}
+
+   Example output: 
+   ```json
+   Rejected due to inappropriate content
```

Comment:
hmm i think the routepolicy might be wrong, bc the intent here is to mask the numbers but not reject them. maybe line 207 to 213 need to be removed, so that only the masking section is left